packages/babel-plugin/src/styled/__tests__/behaviour.test.ts

```diff
@@ -1125,6 +1185,97 @@ describe('styled component behaviour', () => {
     ]);
   });
 
+  it('should apply logical test to class when a conditional branch contains undefined value', () => {
+    const actual = transform(`
+      import { styled } from '@compiled/react';
+
+      const Component = styled.div\`
+        color: \${props => props.isPrimary ? 'green' : undefined};
+      \`;
+    `);
+
+    expect(actual).toIncludeMultiple([
+      '._syazbf54{color:green}',
+      'className={ax(["",props.isPrimary&&"_syazbf54",props.className])}',
+    ]);
+  });
+
+  it('should apply logical test to class when a conditional branch contains null value', () => {
+    const actual = transform(`
+      import { styled } from '@compiled/react';
+
+      const Component = styled.div({
+        color: props => props.isPrimary ? null : 'green',
+      });
+    `);
+
+    expect(actual).toIncludeMultiple([
+      '._syazbf54{color:green}',
+      'className={ax(["",!props.isPrimary&&"_syazbf54",props.className])}',
+    ]);
+  });
+
+  it('should apply logical test to class when a conditional branch contains empty string value', () => {
+    const actual = transform(`
+      import { styled } from '@compiled/react';
+
+      const Component = styled.div({
+        color: props => props.isPrimary ? '' : 'green',
```

Comment:
empty string is useful for ::before or ::after pseudo element.
e.g.
```
div:after {
    content:"";
}
```

Reply:
Good point, we already handle this case, then value will actually be `'""'`, I'll add it to a test case to demonstrate this.